### PR TITLE
Cut 2.4.0 CHANGELOG

### PR DESCRIPTION
**Pull request by staffik**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,13 +3,22 @@
...
 ### Non-protocol Changes
 **No Changes**
 
+## [2.4.0]
+
+### Protocol Changes
+
+* Fixing invalid cost used for `wasm_yield_resume_byte`. [#12192](https://github.com/near/nearcore/pull/12192)
+* Relaxing Congestion Control to allow accepting and buffering more transactions. [#12241](https://github.com/near/nearcore/pull/12241)
+* Exclude contract code out of state witness and distribute it separately. [#11099](https://github.com/near/nearcore/issues/11099)
```
Comment (tayfunelmas) [inline CHANGELOG.md:17]: LGTM. Updated the initial comment with a summary of the changes.

**Review by tayfunelmas — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,13 +3,22 @@
...
 
 ### Non-protocol Changes
 **No Changes**
 
+## [2.4.0]
+
+### Protocol Changes
+
+* Fixing invalid cost used for `wasm_yield_resume_byte`. [#12192](https://github.com/near/nearcore/pull/12192)
+* Relaxing Congestion Control to allow accepting and buffering more transactions. [#12241](https://github.com/near/nearcore/pull/12241)
```
Comment (wacban) [inline CHANGELOG.md:16]: Can you also add the second PR? The latter one overwrites the first one.

**Review by wacban — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,7 +33,6 @@
 ### Non-protocol Changes
 * Added [documentation](./docs/misc/archival_data_recovery.md) and a [reference](./scripts/recover_missing_archival_data.sh) script to recover the data lost in archival nodes at the beginning of 2024.
 * **Archival nodes only:** Stop saving partial chunks to `PartialChunks` column in the Cold DB. Instead, archival nodes will reconstruct partial chunks from the `Chunks` column.
-* Decentralized state sync: Before, nodes that needed to download state (either because they're several epochs behind the chain or because they're going to start producing chunks for a shard they don't currently track) would download them from a centralized GCS bucket. Now, nodes will attempt to download pieces of the state from peers in the network, and only fallback to downloading from GCS if that fails. Please note that in order to participate in providing state parts to peers, your node may generate snapshots of the state. These snapshots should not take too much space, since they're hard links to database files that get cleaned up on every epoch.
```
Comment (staffik) [inline CHANGELOG.md:21]: @VanBarbascu What was I supposed to put there?

Comment (VanBarbascu) [reply CHANGELOG.md:21]: I see that you already moved this part to 2.4.0.
In 2.3.0 we enebled state snapshots on every epoch to allow the nodes to take part in decentralized state sync.

**Pull request merged**